### PR TITLE
Add a periodic alarm every minute as a backup for wake timer

### PR DESCRIPTION
**Pull request by lmorchard**
Not sure if this will absolutely fix alarm reliablilty reports we've
gotten. But, operating on the notion that maybe we're losing the timer
for the next tab that needs to be woken, this commit adds a timer that
fires every minute and causes any pending tabs to wake.

Issue #322.

### CODE REVIEW

**Comment by lmorchard** [comment]
This is just kind of a desperate handwave to make sure tabs wake up. The wake alarm handler checks for pending items and does nothing if none are due to wake, so it shouldn't hurt to call it more often than ideally necessary.

File: src/background.js [hunk 1]
```diff
@@ -269,9 +276,9 @@ function handleWindowCreated(window) {
   }
 }
 
-function handleWake() {
+function handleWake(alarm) {
   const now = Date.now();
-  log('woke at', now);
+  log('woke at', now, 'with alarm', alarm ? alarm.name : 'none');
```
Comment (bwinton) [inline src/background.js]: Could we add some telemetry to see if this catches any alarms that wouldn't have fired?
(I think it's a good idea to add, but would like to know how necessary it is…)

File: src/background.js [hunk 2]
```diff
@@ -260,6 +262,11 @@ function updateWakeAndBookmarks() {
     });
 }
 
+function startPeriodicAlarm() {
+  log('starting periodic alarm');
+  return browser.alarms.create(PERIODIC_ALARM_NAME, { periodInMinutes: 1 });
```
Comment (bwinton) [inline src/background.js]: Will the `return browser.alarms.clearAll()` at line 241/243 clear this alarm, too? And if so, do we need to re-create it in that function?

Comment (lmorchard) [reply src/background.js]: Oof yeah, I think that's right. Will look at that

File: src/background.js (hunk 1 above)
Comment (lmorchard) [reply src/background.js]: Hmm, maybe can think of adding an event for late alarms? Will have to refresh my metrics-fu to figure out how to do taht

File: src/background.js (hunk 2 above)
Comment (lmorchard) [reply src/background.js]: Pushed a tweak to just specifically clear WAKE_ALARM_NAME and not the periodic alarm.

**Comment by lmorchard** [comment]
Going to merge this and submit a follow-up issue to think about adding metrics to measure alarm lateness on wake.

**Pull request merged**